packages/next/src/lib/metadata/resolve-opengraph.ts

```diff
@@ -0,0 +1,88 @@
+import type { Metadata, ResolvedMetadata } from './types/metadata-interface'
+import type { OpenGraphType, OpenGraph } from './types/opengraph-types'
+
+function resolveAsArrayOrUndefined<T = any>(value: T): undefined | any[] {
+  if (typeof value === 'undefined' || value === null) {
+    return undefined
+  }
+  if (Array.isArray(value)) {
+    return value
+  }
+  return [value]
+}
+
+function getFieldsByOgType(ogType: OpenGraphType | undefined) {
+  const articleKeys = ['authors', 'tags'] // article, book
+  const songKeys = ['albums', 'musicians'] // music.song, music.album
+  const playlistKeys = ['albums', 'musicians'] // music.playlist
+  const radioStationsKeys = ['creators']
+  const videoKeys = ['actors', 'directors', 'writers', 'tags']
```

Comment:
extract these out of the function as const arrays. no need to remake them every time this is called